Description: increase test timeout
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2022-12-05

--- a/__tests__/integration/errors.spec.ts
+++ b/__tests__/integration/errors.spec.ts
@@ -8,7 +8,7 @@
 import { findName, genBundle as genBundleH } from "./helpers";
 
 // increase timeout to 15s for whole file since CI occassionally timed out -- these are integration and cache tests, so longer timeout is warranted
-jest.setTimeout(15000);
+jest.setTimeout(300000);
 
 const local = (x: string) => normalize(path.resolve(__dirname, x));
 const testDir = local("__temp/errors");
--- a/__tests__/integration/no-errors.spec.ts
+++ b/__tests__/integration/no-errors.spec.ts
@@ -8,7 +8,7 @@
 import { findName, genBundle as genBundleH } from "./helpers";
 
 // increase timeout to 20s for whole file since CI occassionally timed out -- these are integration and cache tests, so longer timeout is warranted
-jest.setTimeout(20000);
+jest.setTimeout(300000);
 
 const local = (x: string) => path.resolve(__dirname, x);
 const testDir = local("__temp/no-errors");
--- a/__tests__/integration/watch.spec.ts
+++ b/__tests__/integration/watch.spec.ts
@@ -7,7 +7,7 @@
 import * as helpers from "./helpers";
 
 // increase timeout to 15s for whole file since CI occassionally timed out -- these are integration and cache tests, so longer timeout is warranted
-jest.setTimeout(15000);
+jest.setTimeout(300000);
 
 const local = (x: string) => path.resolve(__dirname, x);
 const testDir = local("__temp/watch");
